app/Jobs/GetBlockDetailsJob.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Appeal;
+use App\MwApi\MwApiGetter;
+use App\MwApi\MwApiExtras;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+
+class GetBlockDetailsJob implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    private $appeal;
+
+    public function __construct(Appeal $appeal)
+    {
+        $this->appeal = $appeal;
+    }
+
+    public function handle()
+    {
+        if ($this->appeal->wiki === 'global') {
+            $details = MwApiExtras::getGlobalBlockInfo($this->appeal->appealfor);
+        } else {
+            $details = MwApiExtras::getBlockInfo($this->appeal->wiki, $this->appeal->appealfor);
+        }
+
+        if (!$details) {
+            $this->appeal->update([
+                'status' => 'NOTFOUND',
+            ]);
+
+            return;
```

Comment:
Does this need to send any notifications or add a log entry?

Reply:
No, it would be excessive verbose as it's a system action (you wouldn't believe how many notfounds we get). Devs can still see notfound appeals, and if they fix their block, a fixip entry goes into the logs anyway.